src/constructs/core/stack.test.ts

```diff
@@ -63,4 +64,21 @@ describe("The GuStack construct", () => {
 
     expect(stack.app).toBe("MyApp");
   });
+
+  it("should return a parameter that exists", () => {
+    const stack = new GuStack(new App(), "Test", { app: "MyApp" });
+    const testParam = new GuParameter(stack, "MyTestParam", {});
+    stack.setParam(testParam);
```

Comment:
I'm a bit confused by the interface this is providing.

So is the idea for the client to call `stack.setParam` on parameters they want to make available via `stack.getParam`? I think I spot an opportunity to do the `setParam` in the constructor for `GuParameter` and make all of them available by default, but I suspect I'm actually misunderstanding the usecase

Reply:
> I think I spot an opportunity to do the setParam in the constructor for GuParameter and make all of them available by default

Yeah, that'd be brilliant actually!

> but I suspect I'm actually misunderstanding the usecase

I think I am too 😅 . This change was originally part of https://github.com/guardian/cdk/pull/91, but in that scenario it made more sense for the policy to bring its own parameter with it, rather than get it from the `GuStack`.

I think I misunderstood the API of parameters. I thought they were the same as tags where they had to be [explicitly added](https://github.com/guardian/cdk/blob/7d8fcb2bb0049f0bf3f92f0d36ce52ec6fd304f0/src/constructs/core/stack.ts#L29) to a stack, however it looks like simply declaring a parameter is enough to get it added to the stack.

I think this PR needs more thinking as it certainly feels useful, but not sure for what.

Reply:
Ok, cool! I've moved my review to just a comment rather than "request changes"